tests/aws/services/cloudwatch/test_cloudwatch.py

```diff
@@ -1612,6 +1610,42 @@ def test_describe_minimal_metric_alarm(self, snapshot, aws_client, cleanups):
         response = aws_client.cloudwatch.describe_alarms(AlarmNames=[alarm_name])
         snapshot.match("describe_minimal_metric_alarm", response)
 
+    @markers.aws.validated
+    def test_set_alarm_invalid_input(self, aws_client, snapshot, cleanups):
+        snapshot.add_transformer(snapshot.transform.cloudwatch_api())
+        alarm_name = f"a-{short_uid()}"
+        metric_name = f"m-{short_uid()}"
+        name_space = f"n-sp-{short_uid()}"
+
+        snapshot.add_transformer(TransformerUtility.key_value("MetricName"))
+        aws_client.cloudwatch.put_metric_alarm(
+            AlarmName=alarm_name,
+            MetricName=metric_name,
+            Namespace=name_space,
+            EvaluationPeriods=1,
+            Period=10,
+            Statistic="Sum",
+            ComparisonOperator="GreaterThanThreshold",
+            Threshold=30,
+        )
+        cleanups.append(lambda: aws_client.cloudwatch.delete_alarms(AlarmNames=[alarm_name]))
+        with pytest.raises(Exception) as ex:
+            aws_client.cloudwatch.set_alarm_state(
+                AlarmName=alarm_name, StateValue="INVALID", StateReason="test"
+            )
+
+        err = ex.value.response["Error"]
+        assert err["Code"] == "ValidationError"
+
+        with pytest.raises(Exception) as ex:
+            aws_client.cloudwatch.set_alarm_state(
+                AlarmName=f"{alarm_name}-nonexistent", StateValue="OK", StateReason="test"
+            )
+
+        err = ex.value.response["Error"]
+        assert err["Code"] == "ResourceNotFound"
+        assert not err.get("Message")
```

Comment:
NIT: we could also snapshot the error-responses, e.g 
``` 
snapshot.match("error-resource-not-found", ex.value.response)
```

Reply:
:+1: